server/application/application_event_reporter.go

```diff
@@ -183,13 +184,23 @@ func (s *applicationEventReporter) processResource(
 	manifestGenErr bool,
 	originalApplication *appv1.Application,
 	revisionMetadata *appv1.RevisionMetadata,
+	ignoreResourceCache bool,
 ) {
 	logCtx = logCtx.WithFields(log.Fields{
 		"gvk":      fmt.Sprintf("%s/%s/%s", rs.Group, rs.Version, rs.Kind),
 		"resource": fmt.Sprintf("%s/%s", rs.Namespace, rs.Name),
 	})
 
-	if !s.shouldSendResourceEvent(parentApplication, rs) {
+	if rs.Health == nil && rs.Status == appv1.SyncStatusCodeSynced {
+		// for resources without health status we need to add 'Healthy' status
+		// when they are synced because we might have sent an event with 'Missing'
+		// status earlier and they would be stuck in it if we don't switch to 'Healthy'
+		rs.Health = &appv1.HealthStatus{
+			Status: health.HealthStatusHealthy,
+		}
+	}
```

Comment:
This part is required because sometime resources like ConfigMap get resource `Missing` but after they are healthy they don't have a health status at all which makes the platform stuck on `Missing` health